### PR TITLE
Redis authentication

### PR DESCRIPTION
**Pull request by ihrankouski**
Continued work on #42.
This is makes it possible to enable "requirepass" feature in single node Redis deployment and keep password in k8s Secret.

### CODE REVIEW

File: examples/evaluators/golang/simple/main.go [hunk 1]
```diff
@@ -32,7 +32,9 @@ import (
 	"strings"
 	"time"
 
+	"github.com/GoogleCloudPlatform/open-match/config"
 	om_messages "github.com/GoogleCloudPlatform/open-match/internal/pb"
+	redisHelpers "github.com/GoogleCloudPlatform/open-match/internal/statestorage/redis"
```
Comment (jeremyje) [inline examples/evaluators/golang/simple/main.go]: redishelpers

File: examples/evaluators/golang/simple/main.go [hunk 2]
```diff
@@ -47,30 +49,15 @@ func main() {
...
-	// redis://user:secret@localhost:6379/0?foo=bar&qux=baz // redis pool docs: https://godoc.org/github.com/gomodule/redigo/redis#Pool
-	redisURL := "redis://" + cfg.GetString("REDIS_SERVICE_HOST") + ":" + cfg.GetString("REDIS_SERVICE_PORT")
-	lgr.Println("Connecting to redis at", redisURL)
-	pool := redis.Pool{
-		MaxIdle:     3,
-		MaxActive:   0,
-		IdleTimeout: 60 * time.Second,
-		Dial:        func() (redis.Conn, error) { return redis.DialURL(redisURL) },
-	}
+	pool := redisHelpers.ConnectionPool(cfg)
```
Comment (jeremyje) [inline examples/evaluators/golang/simple/main.go]: This is using the old red ConnectionPool call site. There's an error also sent back.

https://github.com/GoogleCloudPlatform/open-match/pull/87
I think this PR needs to be synced.

Comment (ihrankouski) [reply examples/evaluators/golang/simple/main.go]: yeah I forgot about this, fixed now

**Comment by jeremyje** [comment]
Not Required but useful.

I just added presubmit checks for 040wip. It won't block yet since I don't have it automated yet (coming soon).

Look at [cloudbuild.yaml](https://github.com/GoogleCloudPlatform/open-match/blob/040wip/cloudbuild.yaml#L19) for details. Drop the --substitutions flag, not ready for that yet.

```bash
cloud-build-local --config=cloudbuild.yaml --dryrun=false .

# If you don't have gcloud, install it, https://cloud.google.com/sdk/
gcloud auth login
gcloud components install cloud-build-local
```

**Comment by ihrankouski** [comment]
Yeah I've seen that, it's looking great!
Should I change/uncomment anything in the cloudbuild.yaml before running it?
Just merged latest 040wip changes into this PR and ran `cloud-build-local`, and it finished without errors:
```
...
2019/03/05 12:52:05 status changed to "DONE"
DONE
```
But I can't seem to find the conflicts you mentioned previously...

**Comment by jeremyje** [comment]
> Yeah I've seen that, it's looking great!
> Should I change/uncomment anything in the cloudbuild.yaml before running it?
> Just merged latest 040wip changes into this PR and ran `cloud-build-local`, and it finished without errors:
> 
> ```
> ...
> 2019/03/05 12:52:05 status changed to "DONE"
> DONE
> ```
> But I can't seem to find the conflicts you mentioned previously...

Scroll below this thread. Github is saying it cannot rebase this into the branch because of a conflict. 


> This branch cannot be rebased due to conflicts
> Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

**Comment by jeremyje** [comment]
/gcbrun

I'd love to merge this PR but there's still conflicts. Please sync your branch and resolve.

**Comment by ihrankouski** [comment]
Finally I rebased these changes...

**Pull request merged**